# Homework 2- SOLUTIONS
Using Jupyter and Pandas

In [3]:
# Import pandas module
import pandas as pd

# make plots inline
%matplotlib inline

In [4]:
# Import BabyNames Data
df = pd.read_csv('babynames.csv')
df.head()

Id       Name  Year Gender  Count
0   1       Mary  1880      F   7065
1   2       Anna  1880      F   2604
2   3       Emma  1880      F   2003
3   4  Elizabeth  1880      F   1939
4   5     Minnie  1880      F   1746

## Q1.  What years does this dataset cover?

In [5]:
len(df.Year.unique())

135

**Answer:** 1880 - 2014

## Q2.  How many unique boy names are there in this dataset?

In [6]:
# make a df for girls
df_boy = df[df['Gender'] == "M"]
df_boy.head()

Id     Name  Year Gender  Count
942  943     John  1880      M   9655
943  944  William  1880      M   9532
944  945    James  1880      M   5927
945  946  Charles  1880      M   5348
946  947   George  1880      M   5126

In [7]:
# Use the unique method and take the length of that array
len(df_boy.Name.unique())

39199

**Answer:** there are 39199 unique girl names in this dataset

## Q3.  How many babies were named Taylor in this dataset?

In [11]:
# Here's another way to do it...subset the dataframe using the & sign
# and then sum the Count column
sum(df[(df['Name'] == "Taylor")]['Count'])

416096

**Answer:** there were 416096 babies named Taylor in this dataset

## Q4.  How many more boys were named Logan than girls in 1998?  

In [8]:
# Here's one way to do it...subset the dataframe twice and read off the column
df_boy_alex = df_boy[df_boy['Name'] == "Logan"]
df_boy_alex[df_boy_alex['Year'] == 1998]

Id   Name  Year Gender  Count
1292739  1292740  Logan  1998      M   8639

In [9]:
# do the same for girls
df_girl = df[df['Gender'] == "F"]
df_girl_alex = df_girl[df_girl['Name'] == "Logan"]
df_girl_alex[df_girl_alex['Year'] == 1998]

Id   Name  Year Gender  Count
1276396  1276397  Logan  1998      F   1026

In [10]:
8639 - 1026

7613

**Answer:** there were 7613 more boys named Logan than girls in 1998

## Q5.  What were the three most popular female names in 1998?

In [12]:
df_girl = df[(df.Gender == "F") & (df.Year == 1998)]
df_girl.sort_values(by='Count', ascending=False)[:3]

Id      Name  Year Gender  Count
1276103  1276104     Emily  1998      F  26179
1276104  1276105    Hannah  1998      F  21371
1276105  1276106  Samantha  1998      F  20191

## Q6. 	How many unisex names are in this dataset?  Make a list of the unisex names.

In [14]:
import numpy as np

boyNames = df_boy.Name.unique()
girlNames = df_girl.Name.unique()
np.intersect1d(boyNames, girlNames)

array(['Aaliyah', 'Aareon', 'Aarin', ..., 'Zuriel', 'Zyaire', 'Zyan'], dtype=object)

In [15]:
len(np.intersect1d(boyNames, girlNames))

4453